airsonic-main/src/main/java/org/airsonic/player/spring/HsqlDatabase.java

```diff
@@ -16,4 +16,10 @@ public boolean supportsSchemas() {
             return false;
         }
     }
+
+    @Override
+    public int getPriority() {
+        // to ensure this instance gets selected instead of the parent
```

Comment:
Can you please elaborate on why this is needed?

Reply:
Sure!

Liquibase has a default HsqlDatabase class that it uses internally (which is the parent class of this class)

When we register this class with Liquibase, we need _this_ class to perform migrations for HSQLDB and not _that_ class. Thus we set the priority for this class to be higher (parent + 1) so it gets selected before that class